### PR TITLE
Add paranoid mode

### LINKED ISSUE #1100: Paranoid mode is non existent
While in default configuration here: https://github.com/lynndylanhurley/devise_token_auth/blob/master/config/initializers/devise.rb#L73 it seems that paranoid mode is supported, in fact it isn't at all. I stumbled upon this by accident when I was trying out the password reset api endpoint. It yields different messages depending on whether the email exists in the database or not. With `config.paranoid = true` this behavior isn't changed. I thought it's isolated case, but when I tried to search the repository for `paranoid`, one and only entry was... in that initializer file I mentioned above. Ergo - it's not implemented.

Is this on a roadmap maybe?

 * **Version:** 0.1.42

### PR DESCRIPTION
**Pull request by luisalima**
This PR adds support for Devise paranoid mode and fixes #1100 .
I tried to follow existing conventions from the gem as much as possible, and added a few extra tests for the messages (also when not in paranoid mode).

### CODE REVIEW

File: app/controllers/devise_token_auth/confirmations_controller.rb [hunk 1]
```diff
@@ -53,14 +53,23 @@ def render_create_error_missing_email
     end
 
     def render_create_success
+      message = if Devise.paranoid
```
Comment (MaicolBen) [inline app/controllers/devise_token_auth/confirmations_controller.rb]: I don't like this identation, we waste a lot of space, but if you prefer it this way, we can keep it

File: app/controllers/devise_token_auth/unlocks_controller.rb [hunk 2]
```diff
@@ -61,9 +61,14 @@ def render_create_error_missing_email
     end
 
     def render_create_success
+      message = if Devise.paranoid
```
Comment (MaicolBen) [inline app/controllers/devise_token_auth/unlocks_controller.rb]: same, I see this pattern repeated multiple times, maybe we can create a method that for a I18n key, it returns the paranoid version if `Devise.paranoid` is present

File: app/controllers/devise_token_auth/confirmations_controller.rb (hunk 1 above)
Comment (luisalima) [reply app/controllers/devise_token_auth/confirmations_controller.rb]: @MaicolBen I'm fine with changing it!

File: app/controllers/devise_token_auth/unlocks_controller.rb (hunk 2 above)
Comment (luisalima) [reply app/controllers/devise_token_auth/unlocks_controller.rb]: @MaicolBen thanks for the suggestion, I will do it.

**Comment by luisalima** [comment]
@MaicolBen added the suggested changes. Thank you!

**Comment by luisalima** [comment]
@MaicolBen the build seems to be failing, however it looks like something on the configuration.

File: app/controllers/devise_token_auth/passwords_controller.rb [hunk 3]
```diff
@@ -181,7 +181,11 @@ def password_resource_params
     end
 
     def render_not_found_error
-      render_error(404, I18n.t('devise_token_auth.passwords.user_not_found', email: @email))
+      if Devise.paranoid
+        render_error(404, I18n.t('devise_token_auth.passwords.sended_paranoid'))
```
Comment (ldthorne) [inline app/controllers/devise_token_auth/passwords_controller.rb]: Just as you've updated the response message to indicate success regardless of whether or not it worked, I think this should respond with a 200 to be consistent with the success response.

Comment (ldthorne) [inline app/controllers/devise_token_auth/passwords_controller.rb]: Just as you've updated the response message to indicate success regardless of whether or not it worked, I think this should respond with a 200 to be consistent with the success response.

Comment (luisalima) [reply app/controllers/devise_token_auth/passwords_controller.rb]: Makes sense, apologies for the delay, will get on it now.

**Comment by MaicolBen** [comment]
@luisalima It's failing for mongo, the codeclimate error configuration was already fixed on master

File: app/controllers/devise_token_auth/passwords_controller.rb (hunk 3 above)
Comment (wilkinsondarolt) [reply app/controllers/devise_token_auth/passwords_controller.rb:185]: @luisalima do you need some help to ship this PR?

**Comment by MaicolBen** [comment]
I will merge it after releasing a patch version with some bugfixes

File: app/controllers/devise_token_auth/passwords_controller.rb (hunk 3 above)
Comment (luisalima) [reply app/controllers/devise_token_auth/passwords_controller.rb:185]: @wilkinsondarolt yes most likely... Can't really take it right now :(

**Comment by luisalima** [comment]
Thanks @MaicolBen and sorry for the delay in the response, the pandemic hit hard here.

**Comment by phanyzewski** [comment]
@MaicolBen is there anything left to be done before getting this merged in?

**Comment by devaviary** [comment]
I noticed that with this commit, you can still distinguish between registered emails and unregistered ones in paranoid mode. Like @ldthorne said, the http status needs to be `200` for both cases, and the response message should not be in `errors`. 

Has this been fixed or is anyone working on this already?

**Pull request merged**